### ISSUE TITLE
UBERON terminology update and new terminologies 

### ISSUE THREAD

**Issue by UlrikeS91** [HumanBrainProject/openMINDS_controlledTerms#292]
During an offline meeting, @lzehl and I have discussed some of the latest PRs that we merged. Specifically: issue #278 and PRs #280 & #286. 

Quick summary of the issue:
- we deliberately added "blood" as an organ as a workaround, which is not ideal because there are debates on whether or not blood would count as such
- we added "blood", "arterial blood" and "venous blood" as UBERON parcellations which is also not ideal mainly because the terminology name makes it weird to include them there (they don't fit well as "parcellations" but they are UBERON terms...)
- we generally want to avoid duplicates across terminologies and this is another example where we introduce one (blood is both in "Organ" and "UBEORNParcellation"; this also applies to "brain"...)

Therefore, we would like to revise the setup and propose the following changes:

**TERMINOLOGY RESTRUCTURING:** 
- [Organ](https://github.com/HumanBrainProject/openMINDS_controlledTerms/tree/v1/instances/organ):
     - remains but needs updates
     - this will include "high-level" organ terms (as we already did) but pointing to UBERON as the pref. ontology  
     - TO DO: 
          - [x] rename terminology "organ" to "UBERONOrgan"
          - [x] replace/review existing instances to have UBERON as pref. ontology ID (it's only 2 instances - brain and muscle; and brain can be moved here from the UBERONParcellation terminology)
          - [x] add potential other organs (e.g. "skin", "liver", etc.)
          - [x] (re)move "blood" (to "UBERONBodySubstance")
- UBERONBodySubstance:
     - new terminology
     - this will include anything from body fluids to gases 
     - TO DO:
          - [x] create controlled terms schema for the new terminology [here](https://github.com/HumanBrainProject/openMINDS_controlledTerms/tree/v1/schemas) **with category "study target"** 
          - [x] create terminology instances for the new terminology [here](https://github.com/HumanBrainProject/openMINDS_controlledTerms/tree/v1/instances/terminology)
          - [x] create first instances in a folder for the terminology [here](https://github.com/HumanBrainProject/openMINDS_controlledTerms/tree/v1/instances); first instances could be "feces", "urine", "saliva", "blood", etc.
- [UBERONParcellation](https://github.com/HumanBrainProject/openMINDS_controlledTerms/tree/v1/instances/UBERONParcellation): 
     - remains and mostly unchanged
     - receives a more narrow definition: only includes SUB-structures of organs and still points to UBERON as the pref. ontology
     - TO DO: 
          - [x] remove "blood", "arterial blood" and "venous blood"  
          - [x] (re)move "brain" (to "UBERONOrgan")
          - [x] check the terminology and see if other terms that qualify as organs need to be removed  
          - [x] add potential other substructures of organs (e.g. "hair", "epidermis", "left lobe of the liver", etc.)
- BodySystem:
     - new terminology
     - this will include any system in the body regardless of the scale
     - TO DO:
          - [x] create controlled terms schema for the new terminology [here](https://github.com/HumanBrainProject/openMINDS_controlledTerms/tree/v1/schemas) **with category "study target"** 
          - [x] create terminology instances for the new terminology [here](https://github.com/HumanBrainProject/openMINDS_controlledTerms/tree/v1/instances/terminology)
          - [x] create first instances in a folder for the terminology [here](https://github.com/HumanBrainProject/openMINDS_controlledTerms/tree/v1/instances); first instances could be "GABAergic system", "central nervous system", "peripheral nervous system", etc.
         
**REQUIRED SCHEMA UPDATES:**
 - [TissueSample](https://github.com/HumanBrainProject/openMINDS_core/blob/v4/schemas/research/tissueSample.schema.tpl.json) and [TissueSampleCollection](https://github.com/HumanBrainProject/openMINDS_core/blob/v4/schemas/research/tissueSampleCollection.schema.tpl.json):
 - origin:
      - points to CellType and Organ; remain with some updates
      - TO DO:
           - [x] update "Organ" to "UBERONOrgan" 
           - [x] add "UBERONBodySubstance"
           - [x] adjust instructions accordingly
- anatomicalLocation:
      - points to UBERONParcellation, CustomAnatomicalEntity, ParcellationEntity and ParcellationEntityVersion; remain with some updates
      - TO DO:
           - [ ] add "UBERONOrgan" and "UBERONBodySubstance"

**NOTE:**
This will require a clean-up in the KG!

@lzehl, @ehennestad and @olinux Feel free to add comments or if I forgot anything. If @ehennestad is up for it, you could take this task on 🙂

**Comment by lzehl** [comment]
@UlrikeS91 thanks for preparing this issue. I quickly went over your summary and it looks good. 

As a note for whoever is preparing this: please do not make one big PR but rather a couple of smaller PRs (e.g. one PR to just remove "brain" from UBERONParcellation, one PR to UBERONBodySubstance terminology with an intitial set of substances (e.g. blood stuff) and removing those term from the UBERONParcellation, etc).

Two reasons for smaller PRs: 
1) smaller PRs are quicker reviewed and integrated
2) most of these PRs go along with a cleanup effort within the EBRAINS KG; and also here goes: smaller changes are quicker corrected than bigger ones

**Comment by ehennestad** [comment]
I can get into this on friday, especially the controlled terms. I do not know what is needed in terms of cleaning up the KG, but interested in getting familiar with the process

**Comment by UlrikeS91** [comment]
> I can get into this on friday, especially the controlled terms. I do not know what is needed in terms of cleaning up the KG, but interested in getting familiar with the process

Thank you! The openMINDS restructuring has to happen before the clean-up in the KG anyway. I can give you an intro to what needs to be done there once you complete the other stuff 🙂

**Comment by ehennestad** [comment]
For BodySubstance, is it better to call it OrganismSubstance, to better align with the Uberon term:
https://www.ebi.ac.uk/ols/ontologies/uberon/terms?iri=http%3A%2F%2Fpurl.obolibrary.org%2Fobo%2FUBERON_0000463

**Comment by ehennestad** [comment]
Consider whether BodySystem is the same as AnatomicalSystem:
https://www.ebi.ac.uk/ols/ontologies/uberon/terms?iri=http%3A%2F%2Fpurl.obolibrary.org%2Fobo%2FUBERON_0000467&lang=en&viewMode=PreferredRoots&siblings=false

**Comment by ehennestad** [comment]
UberonParcellations seem to be predominantly parcellations of brain regions and structures of the brain. Would it make sense to make UBERONParcellations a special schema which is brain specific as opposed to substructures of organs. 

Brain anatomy is a field in itself, and it feels a bit wrong to throw a few terms like  "hair" or "epidermis" into a pool of 800+ brain related terms

**Comment by lzehl** [comment]
> Consider whether BodySystem is the same as AnatomicalSystem

@ehennestad I think we originally had in mind for BodySystem to hold all system-like structures of the body including also functional systems not only the anatomical ones (e.g., limbic system). So I would stick to the more generic term. Since it is sometimes a bit difficult to differentiate between or separate anatomical and functional systems I would also not like to split it up.

**Comment by lzehl** [comment]
> UberonParcellations seem to be predominantly parcellations of brain regions and structures of the brain. Would it make sense to make UBERONParcellations a special schema which is brain specific as opposed to substructures of organs.
> 
> Brain anatomy is a field in itself, and it feels a bit wrong to throw a few terms like "hair" or "epidermis" into a pool of 800+ brain related terms

Generally I agree with you about brain anatomy being it's own field. Your suggestion would have though one concrete consequece: if you do this for the brain you need to do this for ALL organs. It could match nicely with the organ terminologies (meaning for each organ terminology there is one parcellation terminology). Please check if that would be feasible. @UlrikeS91 please also comment on this

this could also solve your point about why we add "UBERON" in front. we could change this to "BrainParcellation", "LiverParcellation". etc.

Keep for this point in mind that we have to consider animals in general (not only mammals)

**Comment by UlrikeS91** [comment]
> > UberonParcellations seem to be predominantly parcellations of brain regions and structures of the brain. Would it make sense to make UBERONParcellations a special schema which is brain specific as opposed to substructures of organs.
> > Brain anatomy is a field in itself, and it feels a bit wrong to throw a few terms like "hair" or "epidermis" into a pool of 800+ brain related terms
> 
> Generally I agree with you about brain anatomy being it's own field. Your suggestion would have though one concrete consequece: if you do this for the brain you need to do this for ALL organs. It could match nicely with the organ terminologies (meaning for each organ terminology there is one parcellation terminology). Please check if that would be feasible. @UlrikeS91 please also comment on this
> 
> this could also solve your point about why we add "UBERON" in front. we could change this to "BrainParcellation", "LiverParcellation". etc.
> 
> Keep for this point in mind that we have to consider animals in general (not only mammals)

We had the same issue during our offline discussion and I agree that it feels weird to have all of it mixed, but I'm not sure if we should start terminologies for each organ parcellation. I can imagine that there are cases where the cut isn't 100% clear and it could result in confusion and chaos. This is also the argument for why we only added "molecular entity" and no subdivisions (yet). 
I would like to stick to one terminology for all sub-structures and rather reconsider once we have a library that is large enough to see if any split would make sense (which is again also the plan for "molecular entities").

**Comment by UlrikeS91** [comment]
### Update with changes made/to be made:

**CONTROLLED TERMS UPDATES:**

1. [Organ](https://github.com/HumanBrainProject/openMINDS_controlledTerms/tree/v1/instances/organ):
- remains "Organ" (not renamed to "UBERONOrgan")
- added some additional organs
- removed (or never added, not sure which one) "blood" from organ terminology 
- remaining TO DO:
     - none  
2. OrganismSubstance:
- decided to name it "OrganismSubstance" and not "BodySubstance"
- remaining TO DO:
     - [x] adjust PR #304 according to name change, resolve conflicts and merge
3. [UBERONParcellation](https://github.com/HumanBrainProject/openMINDS_controlledTerms/tree/v1/instances/UBERONParcellation): 
- considered to rename it to "AnatomicalParcellation" but postponed to a later time point
- checked that no other organs are part of terminology (and couldn't find any)
- other substructures of organs can be added on demand, don't need to do this now
- remaining TO DO:
     - [x] adjust PR #304 so that "blood", "arterial blood" and "venous blood" are removed from "UBERONParcellation" 
     - [x] remove "brain" and adjust "organ/brain" with additional information if the uberon term had more/better info
4. [BodySystem](https://github.com/HumanBrainProject/openMINDS_controlledTerms/tree/v1/instances/BodySystem):
- has been added as a new terminology with some first instances
- remaining TO DO:
     - [x] review PR #385 for name change to "OrganismSystem" to be consistet with "OrganismSubstance"

**CORE SCHEMA UPDATES:**
5. [TissueSample](https://github.com/HumanBrainProject/openMINDS_core/blob/v4/schemas/research/tissueSample.schema.tpl.json) and [TissueSampleCollection](https://github.com/HumanBrainProject/openMINDS_core/blob/v4/schemas/research/tissueSampleCollection.schema.tpl.json):
- [x] review PR https://github.com/HumanBrainProject/openMINDS_core/pull/381 once the CT repo has been updated

**SANDS SCHEMA UPDATES:**
6. [ParcellationEntity](https://github.com/HumanBrainProject/openMINDS_SANDS/blob/v3/schemas/atlas/parcellationEntity.schema.tpl.json), [CustomAnatomicalEntity](https://github.com/HumanBrainProject/openMINDS_SANDS/blob/v3/schemas/non-atlas/customAnatomicalEntity.schema.tpl.json) & [AnatomicalTargetPosition](https://github.com/HumanBrainProject/openMINDS_SANDS/blob/v3/schemas/miscellaneous/anatomicalTargetPosition.schema.tpl.json):
- [x] review PR https://github.com/HumanBrainProject/openMINDS_SANDS/pull/212 once the CT repo has been updated

**Comment by lzehl** [comment]
@UlrikeS91 all PRs are merged. I think we tackled all parts of this issue so I will close it now. A big thank you for pushing this!
